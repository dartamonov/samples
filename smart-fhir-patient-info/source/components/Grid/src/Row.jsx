import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

class Row extends Component {

  render() {
    const classes = 'row';

    const {
      children,
      className
    } = this.props;

    return (
      <div className={classNames(className, classes)}>
        {children}
      </div>
    );
  }
}

Row.propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string
};

export default Row;
