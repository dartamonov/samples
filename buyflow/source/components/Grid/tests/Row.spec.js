import React from 'react';
import { create } from 'react-test-renderer';
import Row from '../src/Row.jsx';

describe("Row component", () => {
  it("requires a child component", () => {
    const tree = create(
      <Row><span>Test</span></Row>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
