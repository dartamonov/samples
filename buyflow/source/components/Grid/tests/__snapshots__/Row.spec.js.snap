// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component requires a child component 1`] = `
<div
  className="row"
>
  <span>
    Test
  </span>
</div>
`;
